contracts/UniversalStorageProof_SpokePool.sol

```diff
@@ -0,0 +1,160 @@
+
+    event VerifiedProof(bytes32 indexed dataHash, address caller);
+
+    error NotTarget();
+    error AdminCallAlreadySet();
+    error SlotValueMismatch();
+    error AdminCallNotValidated();
+    error DelegateCallFailed();
+    error AlreadyReceived();
+    error NotImplemented();
+
+    // All calls that have admin privileges must be fired from within the receiveL1State method that validates that
+    // the input data was published on L1 by the HubPool. This input data is then executed on this contract.
+    // This modifier sets the adminCallValidated variable so this condition can be checked in _requireAdminSender().
+    modifier validateInternalCalls() {
+        // Make sure adminCallValidated is set to True only once at beginning of the function, which prevents
+        // the function from being re-entered.
+        if (_adminCallValidated) {
+            revert AdminCallAlreadySet();
+        }
+
+        // This sets a variable indicating that we're now inside a validated call.
+        // Note: this is used by other methods to ensure that this call has been validated by this method and is not
+        // spoofed.
+        _adminCallValidated = true;
+
+        _;
+
+        // Reset adminCallValidated to false to disallow admin calls after this method exits.
+        _adminCallValidated = false;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address _helios,
+        address _hubPoolStore,
+        address _wrappedNativeTokenAddress,
+        uint32 _depositQuoteTimeBuffer,
+        uint32 _fillDeadlineBuffer,
+        IERC20 _l2Usdc,
+        ITokenMessenger _cctpTokenMessenger
+    )
+        SpokePool(_wrappedNativeTokenAddress, _depositQuoteTimeBuffer, _fillDeadlineBuffer)
+        CircleCCTPAdapter(_l2Usdc, _cctpTokenMessenger, CircleDomainIds.Ethereum)
+    {
+        helios = _helios;
+        hubPoolStore = _hubPoolStore;
+    }
+
+    function initialize(
+        uint32 _initialDepositId,
+        address _crossDomainAdmin,
+        address _withdrawalRecipient
+    ) public initializer {
+        __Ownable_init();
+        __SpokePool_init(_initialDepositId, _crossDomainAdmin, _withdrawalRecipient);
+    }
+
+    /**
+     * @notice This can be called by an EOA to relay call data stored on the HubPool on L1 into this contract.
+     * @dev Consider making this an onlyOwner function so that only a privileged EOA can relay state. This EOA
+     * wouldn't be able to tamper with the _publicValues but we can reduce the chance of replay-attacks this way if
+     * we set the EOA to a trusted actor. Replay attacks are possible if this contract has the same address
+     * on multiple chains.
+     * @param _slotKey Slot storage hash
+     * @param _value Slot storage value
+     * @param _blockNumber Block number in light client we want to check slot value of slot key
+     */
+    function receiveL1State(
+        bytes32 _slotKey,
+        bytes calldata _value,
```

Comment:
I can change this to only taking in the message and using address(this) as the target